src/codegen/constant_folding.rs

```diff
@@ -1672,3 +673,1251 @@ fn constants_equal(left: &Expression, right: &Expression) -> bool {
         _ => false,
     }
 }
+
+fn add(
```

Comment:
These functions could use a comment explaining what they do and what their return values mean.